src/metabase/task/sync.clj

```diff
@@ -0,0 +1,115 @@
+(ns metabase.task.sync
+  (:require [clojure.tools.logging :as log]
+            [clojurewerkz.quartzite
+             [conversion :as qc]
+             [jobs :as jobs]
+             [triggers :as triggers]]
+            [clojurewerkz.quartzite.schedule.cron :as cron]
+            [metabase
+             [task :as task]
+             [util :as u]]
+            [metabase.models.database :refer [Database]]
+            [metabase.sfc
+             [analyze :as analyze]
+             [classify :as classify]
+             [fingerprint :as fingerprint]]
+            [toucan.db :as db]))
+
+(def ^:private ^:const classify-databases-job-key     "metabase.task.%s-databases.job-%s")
+(def ^:private ^:const classify-databases-trigger-key "metabase.task.%s-databases.trigger-%s")
+
+(defonce ^:private classify-databases-job (atom nil))
+
+;; simple job which looks up all databases and runs a classify on any saved fingerprints for them
+(jobs/defjob ClassifyDatabase [job-context]
+  (let [db-id    (get (qc/from-job-data job-context) "db-id")
+        database (Database db-id)]
+    (try
+      (log/debug (u/format-color 'green "running scheduled classification for database-id: %s: %s" db-id database))
+      (classify/classify-database! database)
+      (catch Throwable e
+        (log/error (format "Error classifying database %d: (%s)" db-id (:name database)) e)))))
+
+(jobs/defjob AnalyzeDatabase [job-context]
+  (let [db-id    (get (qc/from-job-data job-context) "db-id")
+        database (Database db-id)]
+    (try
+      (log/debug (u/format-color 'green "running scheduled analysis for database-id: %s: %s" db-id database))
+      (analyze/analyze-database! database)
+      (catch Throwable e
+        (log/error (format "Error analyzing database %d: (%s)" db-id (:name database)) e)))))
+
+(jobs/defjob CacheFieldValuesForDatabase [job-context]
+  (let [db-id    (get (qc/from-job-data job-context) "db-id")
+        database (Database db-id)]
+    (try
+      (log/debug (u/format-color 'green "running scheduled caching of field values for database-id: %s: %s" db-id database))
+      (fingerprint/cache-field-values-for-database! database)
+      (catch Throwable e
+        (log/error (format "Error fetching field values for database %d: (%s)" db-id (:name database)) e)))))
+
+(jobs/defjob SyncDatabase [job-context]
+  (let [db-id    (get (qc/from-job-data job-context) "db-id")
+        database (Database db-id)]
+    (try
+      (log/debug (u/format-color 'green "running scheduled caching of field values for database-id: %s: %s" db-id database))
+      (fingerprint/cache-database-field-values! database :full-sync? true)
+      (catch Throwable e
+        (log/error (format "Error fetching field values for database %d: (%s)" db-id (:name database)) e)))))
```

Comment:
@salsakran you wouldn't happen to know what the purpose of this `SyncDatabase` task is, would you? It does basically the same exact thing as the `CacheFieldValuesForDatabase` task, and doesn't actually do any syncing.